streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -91,14 +91,15 @@ private void putAndMaybeForward(final ThreadCache.DirtyEntry entry, final Intern
         try {
             context.setRecordContext(entry.recordContext());
             if (flushListener != null) {
-
                 final V oldValue = sendOldValues ? serdes.valueFrom(underlying.get(entry.key())) : null;
+
+                underlying.put(entry.key(), entry.newValue());
```

Comment:
This is intentional as a bug fix. See description.

Reply:
Wouldn't we want to have an `else` clause that still performs `underlying.put` in case `flushListener` is `null`?

Reply:
Ack, and I think this is the root cause of the jenkins failure.